python/huggingfaceserver/huggingfaceserver/model.py

```diff
@@ -221,6 +227,17 @@ async def generate(self, generate_request: GenerateRequest, headers: Dict[str, s
                 generate_details = Details(finish_reason="length", logprobs=token_outputs)
                 return GenerateResponse(text_output=outputs[0], model_name=self.name, details=generate_details)
 
+    async def completion(self, completion_request: Union[Dict, CompletionRequest], raw_request: [Request], headers: Dict[str, str] = None) \
+            -> Union[CompletionResponse, CompletionStreamResponse, AsyncIterator[Any]]:
+        prompt = completion_request.prompt
+
+        if self.vllm_engine:
+            return await self.openai_serving_completion.create_completion(completion_request, raw_request)
+
+        else:
+            # TODO - fallback flow
+            raise NotImplementedError("completion is not implemented")
```

Comment:
We can still reuse the `generate` code which has already implemented the fallback and replace line 196-199 with `self.openai_serving_completion.create_completion(completion_request, raw_request)`

Reply:
I saw it the other way that we will introduce new method and remove `generate` once this is complete. I can update to edit the same. So with this change `/generate` and `/generate_stream` methods will stop working as previous